subcommand/server-acl-init/command_ent_test.go

```diff
@@ -0,0 +1,436 @@
+				"-enable-namespaces",
+				"-enable-inject-k8s-namespace-mirroring",
+				"-inject-k8s-namespace-mirroring-prefix=prefix-",
+			},
+			AuthMethodExpectedNS:            "default",
+			AuthMethodExpectMirroringConfig: true,
+			BindingRuleExpectedNS:           "default",
+		},
+		"no ns => single dest ns": {
+			FirstRunArgs: nil,
+			SecondRunArgs: []string{
+				"-create-inject-auth-method",
+				"-enable-namespaces",
+				"-consul-inject-destination-namespace=dest",
+			},
+			AuthMethodExpectedNS:            "dest",
+			AuthMethodExpectMirroringConfig: false,
+			BindingRuleExpectedNS:           "dest",
+		},
+		"mirroring ns => single dest ns": {
+			FirstRunArgs: []string{
+				"-enable-namespaces",
+				"-enable-inject-k8s-namespace-mirroring",
+				"-inject-k8s-namespace-mirroring-prefix=prefix-",
+			},
+			SecondRunArgs: []string{
+				"-enable-namespaces",
+				"-consul-inject-destination-namespace=dest",
+			},
+			AuthMethodExpectedNS:            "dest",
+			AuthMethodExpectMirroringConfig: false,
+			BindingRuleExpectedNS:           "dest",
+		},
+		"single dest ns => mirroring ns": {
+			FirstRunArgs: []string{
+				"-enable-namespaces",
+				"-consul-inject-destination-namespace=dest",
+			},
+			SecondRunArgs: []string{
+				"-enable-namespaces",
+				"-enable-inject-k8s-namespace-mirroring",
+				"-inject-k8s-namespace-mirroring-prefix=prefix-",
+			},
+			AuthMethodExpectedNS:            "default",
+			AuthMethodExpectMirroringConfig: true,
+			BindingRuleExpectedNS:           "default",
+		},
+		"mirroring ns => mirroring ns": {
+			FirstRunArgs: []string{
+				"-enable-namespaces",
+				"-enable-inject-k8s-namespace-mirroring",
+				"-inject-k8s-namespace-mirroring-prefix=prefix-",
+			},
+			SecondRunArgs: []string{
+				"-enable-namespaces",
+				"-enable-inject-k8s-namespace-mirroring",
+				"-inject-k8s-namespace-mirroring-prefix=prefix-",
+			},
+			AuthMethodExpectedNS:            "default",
+			AuthMethodExpectMirroringConfig: true,
+			BindingRuleExpectedNS:           "default",
+		},
+	}
+
+	for name, c := range cases {
+		t.Run(name, func(tt *testing.T) {
+			require := require.New(tt)
+			k8s, testAgent := completeEnterpriseSetup(tt, resourcePrefix)
+			defer testAgent.Stop()
+			setupK8sServiceAccount(tt, k8s)
+
+			ui := cli.NewMockUi()
+			defaultArgs := []string{
+				"-server-label-selector=component=server,app=consul,release=" + releaseName,
+				"-resource-prefix=" + resourcePrefix,
+				"-k8s-namespace=" + ns,
+				"-expected-replicas=1",
+				"-create-inject-auth-method",
+			}
+
+			// First run. NOTE: we don't assert anything here since we've
+			// tested these results in other tests. What we care about here
+			// is the result after the second fun.
+			cmd := Command{
+				UI:        ui,
+				clientset: k8s,
+			}
+			responseCode := cmd.Run(append(defaultArgs, c.FirstRunArgs...))
+			require.Equal(0, responseCode, ui.ErrorWriter.String())
+
+			// Second run.
+			cmd = Command{
```

Comment:
Does this need to be declared again?

Reply:
See comment above. In this case it has to be otherwise the old flags stay set.